### ISSUE TITLE
error while loading shared libraries: libgstapp-0.10.so.0

### ISSUE THREAD

**Issue by phanorcoll** [toggl/toggldesktop#2165]
Im trying to run ./TogglDesktop.sh on Fedora25 and I get

error while loading shared libraries: libgstapp-0.10.so.0: cannot open shared object file: No such file or directory

and running ./TogglDesktop I get

./TogglDesktop: error while loading shared libraries: libTogglDesktopLibrary.so.1: cannot open shared object file: No such file or directory

Is there a way to make it work on linux.. Thanks

**Comment by IndrekV** [comment]
Seems similar issue to #1894 

Try installing the missing dependencies with:
```
sudo apt-get install libgstreamer-plugins-base0.10-0
```

**Comment by fonorobert** [comment]
@IndrekV multiple users reported this under Fedora 25 and 26. Note that apt-get is the package manager for Ubuntu, the needed version of libgstreamer doesn't seem to be in the repos there.

**Comment by fonorobert** [comment]
@IndrekV users reporting this from under Solus as well (eg. case/520243). this outdated version of the gstreamer libraries isn't in the repos there either.
Currently it seems more and more like for anyone not on Ubuntu it's a question of luck whether they can run the Toggl app or not, depending on their distro, distro version, etc.

Updating to a more up-to-date version of QT would resolve this for all the distros I've seen with this issue so far.

**Comment by fonorobert** [comment]
more info on the Solus case: Solus does have gstreamer 0.10 in the repos, user installed it but when running the .sh they're still getting an error about libstapp-0.10.s.0 missing

**Comment by naroga** [comment]
+1, I too am having this issue.

**Comment by naroga** [comment]
@fonorobert `sudo dnf install gstreamer gstreamer-plugins-base-devel` seems to workaround this issue.

**Comment by mikeadamz** [comment]
Same problem on KaOS 2017.07. It comes with Gstreamer 1.12.2

**Comment by mmrmartin** [comment]
`gstreamer gstreamer-plugins-base-devel` didn't helped me on openSUSE Tumbleweed 20170913. OpenSUSE has no more gstreamer 0.10 in repository, only has 1.12 version...

**Comment by kk-1** [comment]
I got openSUSE Tumbleweed 20170913 too and use 64 bit (all in 64 bit!!!) version of rstudio
I tried to create symlinks (DON'T TRY THEM. I ALREADY DELETED THEM) :

ln -s /usr/lib64/libgstreamer-1.0.so.0.1203.0  /usr/lib64/libgstreamer-0.10.so.0
ln -s /usr/lib64/libgstapp-1.0.so.0  /usr/lib64/libgstreamer-0.10.so.0
ln -s /usr/lib64/libgstbase-1.0.so.0.1203.0  /usr/lib64/libgstbase-0.10.so.0

The first three get me somewhere yet then I could not find any instance of a:
libgstinterfaces-0.10.so.0

in the below repo there are gstreamer-0.10 stuff yet then asks liborc-test!!! :

https://download.opensuse.org/repositories/home:/tdoerges/openSUSE_Tumbleweed/x86_64/



By the way software search of openSUSE shows nonexistent files (at least I could not find them!)
https://download.opensuse.org/repositories/multimedia:/libs/openSUSE_Tumbleweed/x86_64/libgstapp-0_10-0-0.10.36-88.7.x86_64.rpm


In short could not fix it!!!

Definitely, rstudio needs an update!!!

**Comment by martinchodev** [comment]
I had the same problem in Manjaro Linux. Installing `gstreamer0.10-base-plugins` like this fixed the problem:

`$ yaourt -S gstreamer0.10-base-plugins`

**Comment by ctrlaltdev** [comment]
> @fonorobert `sudo dnf install gstreamer gstreamer-plugins-base-devel` seems to workaround this issue.
> 
> Tested on Fedora 26.

I tested on Fedora 28, it works too.
Could you add the workaround to the Linux Support Page (https://support.toggl.com/toggl-desktop-for-linux/)?

**Comment by szczerb** [comment]
Any news on that? I just tried to run the latest tarball on Gentoo and I get the same exact error.

Old gstreamer is currently hard-masked on Gentoo since a few weeks ago.

I'd prefer not to have to install it as there are ove 20 CVE vulnurabilites it's affected by and actually the move to get rid of it started back in 2015 because "it is dead and unmaintained for ages".
https://bugs.gentoo.org/550648

**Comment by MartinBriza** [comment]
In Gentoo, I recommend compiling TogglDesktop from the commandline. I see a Gentoo ebuild that has not been maintained for quite a bit. I'll try contacting the maintainer to update it so you don't have to resort for the tarball.
I'll also look into the GStreamer issue, so we don't rely on outdated libraries in the tarball.

**Comment by szczerb** [comment]
Could you point me to that old ebuild? It's not in the tree and I can't find it in the overlays as well.

Thanks.

**Comment by brunommpreto** [comment]
Same issue on Opensuse Tumbleeweed 5.2.5-1-default

My work around was installing toggl through flatpak. The is also available in snapcraft

Via flatpak : https://flathub.org/apps/details/com.toggl.TogglDesktop
Via snap: https://snapcraft.io/toggldesktop
